### PR TITLE
Fix extra metrics tracking and eval dataset size issues

### PR DESCRIPTION
**Pull request by tmfs10**

### CODE REVIEW

File: openrlhf/trainer/ppo_trainer.py [hunk 1]
```diff
@@ -380,8 +383,9 @@ def training_step(self, experience: Experience, global_steps) -> Dict[str, float
         return status
 
     def training_step_actor(self, experience: Experience) -> Dict[str, float]:
-        self.actor.train()
         start_time = time.time()
+        """
+        self.actor.train()
```
Comment (gwarmstrong) [inline openrlhf/trainer/ppo_trainer.py:388]: what does commenting this out achieve?

File: openrlhf/trainer/ppo_trainer.py [hunk 2]
```diff
@@ -363,7 +363,10 @@ def ppo_train(self, global_steps=0):
                     status_mean[k] += v
             for k in status_mean.keys():
                 if type(status_mean[k]) == list:
-                    status_mean[k] = torch.cat(status_mean[k]).mean().item()
+                    s = torch.cat(status_mean[k])
+                    mask = s != -99999
```
Comment (gwarmstrong) [inline openrlhf/trainer/ppo_trainer.py:368]: is the correct interpretation here that your reward function is passing back values of -99999 for things that should be masked? If so, can you please make a corresponding comment here.

**Review by gwarmstrong — COMMENTED** [review]
couple minor comments/questions

File: openrlhf/trainer/ppo_trainer.py (hunk 1 above)
Comment (titu1994) [reply openrlhf/trainer/ppo_trainer.py:388]: This should be reverted

Comment (tmfs10) [reply openrlhf/trainer/ppo_trainer.py:388]: Oh yea my mistake

Comment (tmfs10) [reply openrlhf/trainer/ppo_trainer.py:388]: Was put in as part of debugging

File: openrlhf/trainer/ppo_trainer.py (hunk 2 above)
Comment (tmfs10) [reply openrlhf/trainer/ppo_trainer.py:368]: Yes that is the case. Will make that comment

**Comment by tmfs10** [comment]
@gwarmstrong can you take a final look? Pushed the fixes

File: openrlhf/trainer/ppo_trainer.py [hunk 3]
```diff
@@ -554,7 +558,14 @@ def evaluate(self, dataloader, global_step, extra_rm_args):
         for k, v in eval_buffer.items():
             if len(eval_buffer[k]) > 0:
                 metrics = torch.cat(eval_buffer[k])
-                logs_dict[k] = metrics.mean().item()
+
+                if k.startswith('metric_'):
+                    # -99999 values are masked and ignored
```
Comment (gwarmstrong) [inline openrlhf/trainer/ppo_trainer.py]: this comment just explains what it is clear from the line of code. please provide the context for the code (e.g., "we decided to use a sentinel value of -99999 to mask reward values from the server")

Comment (tmfs10) [reply openrlhf/trainer/ppo_trainer.py]: How's the update?

Comment (tmfs10) [reply openrlhf/trainer/ppo_trainer.py]: @gwarmstrong

File: openrlhf/trainer/ppo_trainer.py [hunk 4]
```diff
@@ -363,7 +363,11 @@ def ppo_train(self, global_steps=0):
                     status_mean[k] += v
             for k in status_mean.keys():
                 if type(status_mean[k]) == list:
-                    status_mean[k] = torch.cat(status_mean[k]).mean().item()
+                    s = torch.cat(status_mean[k])
+                    # -99999 values (sentinel values) for the keys with "metrics_" prefix (not the rewards) are masked
```
Comment (gwarmstrong) [inline openrlhf/trainer/ppo_trainer.py]:
```suggestion
                    # -99999 values (sentinel values) for the keys with "metrics_" prefix are masked
```

**Comment by tmfs10** [comment]
@gwarmstrong Good with everything else? If so I'll merge

**Review by gwarmstrong — APPROVED** [review]
LGTM
